src/msw/thread.cpp

```diff
@@ -1121,6 +1122,96 @@ wxThreadError wxThread::Kill()
     return rc;
 }
 
+// At least MSVC 2017 version 15.6 is required for observing the
+// thread names set using this method.
+// Windows 10 version 1607 is required for the SetThreadDescription
+// function.
+static bool wxSetThreadNameOnWindows10(const WCHAR *threadName)
+{
+    typedef HRESULT(WINAPI* SetThreadDescription_t)(HANDLE, PCWSTR);
+    static SetThreadDescription_t s_pfnSetThreadDescription = NULL;
+
+    static bool s_initDone = false;
+    if ( !s_initDone )
+    {
+        wxLoadedDLL dllKernel32("kernel32.dll");
+        wxDL_INIT_FUNC(s_pfn, SetThreadDescription, dllKernel32);
+        s_initDone = true;
+    }
+
+    if ( s_pfnSetThreadDescription ) {
+        HRESULT r = (*s_pfnSetThreadDescription)
```

Comment:
I may be missing something, why not just write 
`r = s_pfnSetThreadDescription(GetCurrentThread(), threadName);`

The opening brace above also should be on its own line, BTW.

Reply:
I happened to base my dynamic loading code on what I found in src\common\translation.cpp, and for some reason that very file uses the more complicated syntax.